cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/Language.kt

```diff
@@ -246,54 +263,116 @@ abstract class Language<T : LanguageFrontend<*, *>> : Node() {
 
         // Check, if super type (or its root) is in the list. Also, the wrap state needs to be the
         // same
-        return superType.root in superTypes && type.wrapState == superType.wrapState
+        return targetType.root in superTypes && type.wrapState == targetType.wrapState
+    }
+
+    open fun tryCast(
+        type: Type,
+        targetType: Type,
+        hint: HasType? = null,
+        targetHint: HasType? = null
+    ): CastResult {
+        // We can take a shortcut if it is the same type
+        if (type == targetType) {
+            return DirectMatch
+        }
+
+        // We can also take a shortcut: if they are not of the same subclass, they will never
+        // match
+        if (type::class != targetType::class) {
+            return CastNotPossible
```

Comment:
Are there no casts where our type classes can differ? I intuitively think of wrappers for primitive types in java but I don't know if our modeling currently has an issue with the casting here.

Reply:
Unboxing of primitive types in Java is probably something that needs to be specified in the Java language